core/domain/question_services_test.py

```diff
@@ -87,6 +97,24 @@ def test_get_questions_by_skill_ids(self):
         self.assertEqual(
             questions[0].to_dict(), self.question.to_dict())
 
+    def test_get_questions_by_skill_ids(self):
+        question_services.create_new_question_skill_link(
+            self.question_id, 'skill_1', 0.3)
+        question_services.create_new_question_skill_link(
+            self.question_id_1, 'skill_2', 0.8)
+        question_services.create_new_question_skill_link(
+            self.question_id_2, 'skill_2', 0.5)
+
+        questions = question_services.get_questions_by_skill_ids(
+            4, ['skill_1', 'skill_2'])
+        question_dicts = [
+            self.question.to_dict(), self.question_1.to_dict(),
+            self.question_2.to_dict()
+        ]
+        self.assertEqual(len(questions), 3)
+        for question in questions:
+            self.assertTrue(question.to_dict() in question_dicts)
```

Comment:
This is weak, maybe better to collect all the dicts and then sort both lists and compare them.

Reply:
Done.